### ISSUE TITLE
[JOSS REVIEW] Additional Paper Feedback

### ISSUE THREAD

**Issue by JHartzer** [auctus-team/pycapacity#9]
# General Feedback
As mentioned in #6, I would suggest not saying the software is real-time as there are no guarantees that the software will produce calculations within a certain time frame. Moreover, it is technically possible for system interrupts to cause a python function call to take *much* longer than the times listed because of various OS functions. Because you mention safety applications in your paper, I would stress that this software is at best *near-real-time* or *online*. However, as you still would like to mention the performance, timing variance or distribution would be useful to add.

Additionally, I believe there should be a mention of the rate at which the functions are called. The execution times are useful to see, but if I need to call the Acceleration polytope metric more than 3 times per second, then this software is no longer even near real-time. It is my understanding that inverse kinematics and other constraints are calculated tens or hundreds of times per second and I wonder if this would be a limitation when considering musculoskeletal models.

There should be additional discussion and citations on the current state of the field in regards to alternative software solutions and where this work differs.

Lastly, the paper ends very abruptly. The addition of concluding remarks would help greatly as the introduction and statement of need are already strong.

## Introduction
- "where he lacks" use the "they"
- It can be interesting ~~to be able~~ to compute these metrics

## Statement of Need
- "real-time" should be updated to "near real-time" or "online"
- "aslo" -> "also"

## Iterative convex-hull method
- Remove "And" from the start of the third sentence and remove the paragraph break

## Robotic manipulator metrics
- "For ~~the~~ robotics manipulators, the package integrates"

## Polytopes
- Move "This approach is described by..." to the end of the following paragraph

## Polytope evaluation algorithms used
- Create captions for the performance tables
- Reference the table itself in the text, similar to how the figures were done

**Comment by askuric** [comment]
Hi @JHartzer,

Thanks for your suggestions. I've integrated most of them to the paper. It is heavily modified so I'm looking forward to your feedback.

Here is a quick resume of the changes:
- a bit more consequential Statement of need with a discussion about the existing solutions
- a bit more info about what is the main aim of the library (apart form the near real-time)
- I've done a bit more benchmarking and updated the table with the statistics, as well as provided the benchmarking scripts in the examples folder so that everyone can check on their system
- Added the conclusion
- A section about the differences of ellipsoids and polytopes, and what do they actually represent added 

I've written a bit more in detail in #6 .


>As mentioned in https://github.com/auctus-team/pycapacity/issues/6, I would suggest not saying the software is real-time as there are no guarantees that the software will produce calculations within a certain time frame. Moreover, it is technically possible for system interrupts to cause a python function call to take much longer than the times listed because of various OS functions. Because you mention safety applications in your paper, I would stress that this software is at best near-real-time or online. However, as you still would like to mention the performance, timing variance or distribution would be useful to add.

You are right. I've refactored our claims in the paper to put less accent on the real-time capability and more on the ease of use and flexibility. Also I've added more finely measured execution times and the variances and included the benchmarking scripts in the package examples so that everyone can test the performance of their system and make sure its suitable for their application.

> Additionally, I believe there should be a mention of the rate at which the functions are called. The execution times are useful to see, but if I need to call the Acceleration polytope metric more than 3 times per second, then this software is no longer even near real-time. It is my understanding that inverse kinematics and other constraints are calculated tens or hundreds of times per second and I wonder if this would be a limitation when considering musculoskeletal models.

These metrics are not necessarily intended to be calculated thousands of times per second. Especially then it comes to using musculoskeletal models, the biomechanics software is often slow, and in case of OpenSim, to obtain one of the system matrices that is needed for Acceleration polytope calculation it can take up to one second . So to answer your question, it depends of the application. 

In our case we've used this package for robot control and we've run the human and robot force polytope calculation in the 50-100Hz range, which was good enough for our case, but might not be suitable for others.

But the goal of talking about real-time was to communicate that these polytope based metrics are not as slow as they are considered to be and that they can be used in online interactive manner in some cases. And the goal of the package is to provide a simple and flexible way of using them. 

In the next version of the paper, I've tried to communicate this a bit clearer.
